hicommonwealth/commonwealth
# Community Redirects

## Link to Issue
Closes: #6863

## Description of Changes
- Adds the `Communities.redirect` column which can contain an old community id
- Adds `app.config.redirects` which aggregates any defined `Communities.redirect` values
- Adds checks in create community flows on client + API to ensure the new community id does not match any of the redirect values.

## Test Plan
1. Navigate to admin-panel
2. Update any communities id
3. Access the old community id directly by URL (ensure you are properly redirected
4. Click on a thread in the new community
5. Edit URL -> change new community id to old community id and ensure redirect to thread works
6. Search for the old community in the search bar and ensure it doesn't show up
7. Search for the new community in the search bar and ensure it only shows up once
8. Enabled community directory for the same chain node as the community whose id you updated
9. Search for the old/new community ids in the directory

## Deployment Plan

1. 

## Other Considerations

-
